app/models/report.rb

```diff
@@ -46,4 +46,11 @@ def end_date
       start_date + duration.days
     end
   end
+
+  private
+
+  def active_components(opts={})
+    result = text_components.includes(:channels)
+    text_components.select {|c| c.active?(opts) }
+  end
```

Comment:
good point. Is there any situation where we want bot `active` and `unpublished` text components? I'm going to change this..